File: PYTHON PROJECT/db/p2.py
import cx_Oracle
con=None
cursor=None
try:
	con=cx_Oracle.connect("system/abc123")
	print("connected")
	cursor=con.cursor()
	sql="insert into employee values('%d','%s')"
	eid=int(input("enter employee id "))
	ename=input("enter employee name ")
	args=(eid,ename)
	cursor.execute(sql % args)
	con.commit()
	print(cursor.rowcount,"records inserted")

except cx_oracle.DatabaseError as e:
	con.rollback()
	print("issue",e)
finally:
	if cursor is not None:
		cursor.close()
	if con is not None:
		con.close()
		print("disconnected")

File: JAVA LEC/L20/P2.java
import java.io.*;
import java.sql.*;

class P2
{
public static void main(String args[])
{
Console c=System.console();

try
{
DriverManager.registerDriver(new oracle.jdbc.driver.OracleDriver());
Connection con=DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:xe","system","abc123");
System.out.print("Connected");

String  s1="insert into employee values(?,?)";
PreparedStatement s2=con.prepareStatement(s1);

int eid=Integer.parseInt(c.readLine("enter emp id "));
String name=c.readLine("enter emp name ");

s2.setInt(1,eid);
s2.setString(2,name);

int r1=s2.executeUpdate();
System.out.println(r1 + "records inserted ");

con.close();
System.out.println("DisConnected");
}
catch(SQLException se)
{
System.out.println("issue"+se);
}
}
}
